src/DynamoCore/Logging/AnalyticsService.cs

```diff
@@ -65,9 +68,10 @@ internal static bool IsADPOptedIn
                 {
                     return false;
                 }
-
-                return adpAnalyticsUI.IsOptedIn();
+                //TODO add this overload to IAnalyticsUI.
+                return (adpAnalyticsUI as ADPAnalyticsUI ).IsOptedIn(30000);
```

Comment:
if you add a reference to `ADPAnalyticsUI` here...then, while calling  `IsADPOptedIn` the ADP assembly will be loaded even if `DisableAnalytics` will be set to true.
DotNet resolves its types when parsing the function body...even if the type's function/property is never called ...

Reply:
So you are saying by using this `IAnalyticsUI` we can avoid loading the ADP binary when `DisableAnalytics` - so if I complete this TODO everything is fine?

Are there tests that assert when `DisableAnalytics` is true that the binaries are not loaded? It's hard to reason about otherwise.

Reply:
`so if I complete this TODO everything is fine` I think so
`Are there tests that assert when DisableAnalytics is true that the binaries are not loaded? ` I'll take a look. I think I can add it if it does not already exist